### DISCUSSION TITLE
Segredos do GitHub Pages: Funcionalidades e Truques Escondidos

### DISCUSSION THREAD

**Discussion by 0xstrideer** [community/community#134211] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### **Temas Personalizados com Jekyll:**
Embora o GitHub Pages ofereça temas padrão para sites Jekyll, você pode criar ou usar temas personalizados. Basta clonar o repositório do tema desejado e personalizá-lo de acordo com suas necessidades. Isso permite que você tenha um design único sem se limitar às opções padrão

### **Integração de Domínios Personalizados:**
Você pode usar seu próprio domínio para seu site no GitHub Pages. Para isso, adicione um arquivo CNAME no diretório raiz do seu repositório com o nome do domínio. Em seguida, configure o registro DNS do seu domínio para apontar para o GitHub Pages

### **Proteção com HTTPS:**
GitHub Pages oferece HTTPS gratuito para sites com domínio personalizado, garantindo segurança e confiança para os visitantes. Basta habilitar essa opção nas configurações do repositório em "GitHub Pages" e aguardar a configuração automática

### **Automatização com GitHub Actions:**
Use o GitHub Actions para automatizar a geração e implantação do seu site. Você pode configurar workflows que automaticamente constroem e publicam o site toda vez que um commit é feito no repositório

### **Uso de Plugins Jekyll:**
Embora o GitHub Pages tenha restrições quanto a plugins Jekyll, você pode usar GitHub Actions para construir o site localmente com plugins e, em seguida, enviar os arquivos estáticos resultantes para o branch do GitHub Pages

### **Versão Local com Jekyll:**
Para testar seu site localmente antes de publicar, você pode instalar o Jekyll em sua máquina e executar jekyll serve. Isso permite visualizar e depurar o site em seu navegador

### **Compatibilidade com SPA (Single Page Applications):**
É possível hospedar SPAs no GitHub Pages. Use arquivos .htaccess ou 404.html para redirecionar todas as requisições para o index.html e garantir o funcionamento adequado das rotas

**Comment by striderws** [comment]
Thank you for the tips
